src/capstone.rs

```diff
@@ -157,7 +154,7 @@ impl<'cs> Capstone<'cs> {
 
     /// Disassemble all instructions in buffer
     pub fn disasm_all<'a>(
-        &mut self,
```

Comment:
I think this is like a UnsafeCell hidden behind FFI... So if we can prove that it is not possible to have two `disasm` call in progress simultaneously, we should be safe.

Reply:
On the other hand, if this turn out not safe, then the `csh` method will need to accept `&mut self`, as the semantic requires `csh` works like a `*mut c_void`.

Reply:
As `Send` or `Sync` is not an issue, unwinding is become the main focus as this is another way to run code simutaneously (i.e. mutation in one piece of code panic, and unwind calls another piece of mutating code).

So your unsafe advanture will arrive the next stop: unwind safety. The following is from Nomicon:

> Rust's unwinding strategy is not specified to be fundamentally compatible with any other language's unwinding. As such, unwinding into Rust from another language, or unwinding into another language from Rust is Undefined Behavior. You must absolutely catch any panics at the FFI boundary! What you do at that point is up to you, but something must be done. If you fail to do this, at best, your application will crash and burn. At worst, your application won't crash and burn, and will proceed with completely clobbered state.

Reply:
So, unfortunately there is no easy solutions, you need to be ready to create a wrapper in C and making sure all exceptions are handled in the wrapper. Or, if you are sure the C function is already exception free, you can just trust it.

Reply:
In this library, the Rust code calls into the C code; the C code never calls into the Rust (or any other language).
From that point of view, we should also be safe (since C has no exceptions or stack unwinding).